pkg/util/iptables/iptables.go

```diff
@@ -285,7 +285,15 @@ func (runner *runner) DeleteChain(table Table, chain Chain) error {
 	runner.mu.Lock()
 	defer runner.mu.Unlock()
 
-	// TODO: we could call iptables -S first, ignore the output and check for non-zero return (more like DeleteRule)
```

Comment:
I don't understand this TODO, also the interface doc says:

> 	// DeleteChain deletes the specified chain.  If the chain did not exist, return error.
>	DeleteChain(table Table, chain Chain) error

that is something that current iptables operation -X already does

Reply:
I found that where the delete function was called, the error was handled
for example:
https://github.com/kubernetes/kubernetes/blob/991bb65ecc06a9e388f411f5a3f9a8737844ea70/pkg/proxy/ipvs/proxier.go#L790-L791

so I thought it's ok to delete this todo comment, do you think so?

Reply:
yes, but better @danwinship or @thockin ACK for confirmation

Reply:
I only vaguely recall this context :)

It seems like it would be simpler to just look at the Error() for "No chain/target/match by that name" ?

Reply:
@thockin yes,  there are two ways here. The first is just look at the Error() for "No chain/target/match by that name". if chain do not exists, the return error is "No chain/target/match by that name".

the second is call `iptables -S`, just like done in pr, and we don’t need to call the `utiliptables.IsNotFoundError` to look the error. 

So all of that to say, I think the first way is better. just delete this todo comment

Reply:
I agree it is simpler to just do it and check the error.